### DISCUSSION TITLE
Direct2D1 wave pattern with animated turbulence offset

### DISCUSSION THREAD

**Discussion by Odepax** [amerkoleci/Vortice.Windows#391] [Q&A]
I'm trying to reproduce the following effect using Direct2D1:

![](https://community.adobe.com/t5/image/serverpage/image-id/122838i6C67D5602E969800/image-size/medium?v=v2&px=400)

I thought it would be a _"simple"_ case of using the [built-in turbulence effect](https://learn.microsoft.com/en-us/windows/win32/direct2d/turbulence), and adjusting the offset property, as the documentation says:

> Offset\
> `D2D1_TURBULENCE_PROP_OFFSET`
>
> The coordinates where the turbulence output is generated. The algorithm used to generate the Perlin noise is position dependent, so a different offset results in a different output. This property is not bounded and the units are specified in DIPs.
>
> Note: **The offset does not have the same effect as a translation because the noise function output is infinite and the function will wrap around the tile.**

So I coded this simple turbulence effect:

```cs
public void OnRender(ID2D1DeviceContext g, float t) {
   using var turbulenceEffect = new Turbulence(g) {
      Offset = new Vector2(t, 0)
   };

   g.Clear(Colors.Black);
   g.DrawImage(turbulenceEffect);
}
```

However, the render outputs a turbulence image slowly crawling rightward, acting as if the turbulence offset was indeed a translation, unlike what I understand from the documentation quoted above.

Could it be a bug in Vortice? Is my understanding of the docs incorrect? If not, what would be the correct way to implement the desired effect?

* * *

[Reference image source](https://community.adobe.com/t5/after-effects-discussions/fractal-noise-s-offset-turbulence-has-maximum-value-how-can-i-keep-going/m-p/11302493#M117629)\
[Vortice.Direct2D1](https://www.nuget.org/packages/Vortice.Direct2D1/) 2.4.2\
.NET Target Framework: `net6.0-windows10.0.19041.0`

**Comment by amerkoleci** [comment]
Try to reproduce it in native and see how it behaves, then we can make a conclusion

**Reply by Odepax** [reply]
I doubt I'll have the competence to go anywhere with native. 😢

That said, [SO](https://stackoverflow.com/questions/75976544/direct2d1-wave-pattern-with-animated-turbulence-offset?noredirect=1#comment134006599_75976544) suggests that my understanding of the doc is incorrect.
